ui/analyse/src/util.ts

```diff
@@ -112,8 +113,13 @@ function toLink(url: string) {
   return '<a target="_blank" rel="nofollow" href="' + url + '">' + show + '</a>';
 }
 
-export function enrichText(text: string, allowNewlines: boolean): string {
-  let html = autolink(window.lichess.escapeHtml(text), toLink);
+const markdownLinkRegex = /\[([^<\]]+)\]\(((?:(?:https?|ftp):\/\/[\-A-Z0-9+\u0026\u2019@#\/%?=()~_|!:,.;]*[\-A-Z0-9+\u0026@#\/%=~()_|]))\)/gi;
```

Comment:
Very minor issue: The regex excludes URLs like `http://example.com?`, because in normal text the `?` should not be part of the link.

Reply:
Do URLs ever end with a question mark without anything following? `http://example.com?test=1` gets matched fine, for example.